docs/concepts/manifest-builds.md

```diff
@@ -0,0 +1,169 @@
+---
+title: "Preview: Builds"
+description: How to use Flox environments to build artifacts 
+---
+
+The typical development lifecycle involves a step where your source code and
+potentially some other assets are bundled together into an artifact.
+That could be a compiled executable, an archive containing source files, or
+something else entirely.
+A Flox environment ensures that the same set of tools, dependencies, and
+environment variables are available wherever the environment is activated.
+You can now integrate builds into your environment to build those artifacts
+with the same reproducible set of tools and dependencies that were used to
+develop the software.
+
+## Enabling builds
+
+Builds are still in development,
+so before you can perform builds you will need to enable a feature flag.
+
+To enable the builds feature temporarily you can set an environment variable:
+
+```bash
+$ export FLOX_FEATURES_BUILD=true
+```
+
+To make this feature flag persistent you can edit your Flox config:
+
+```bash
+$ flox config --set-bool features.build true
+```
+
+## Defining builds
+
+Builds are defined in the `[build]` section of the manifest.
+Configuring a build mostly entails providing a short Bash script to run to
+build the artifact.
+This script is run inside the environment so that the tools used to develop
+the software are available during the build.
+
+Each build specified in the `[build]` section corresponds to a different
+artifact.
+This allows you to produce multiple artifacts from a given set of sources,
+for example to produce different versions of a compiled binary both with
+and without debug symbols.
+
+Finally, `flox` requires that you place your artifacts into a directory called
+`$out`,
+but we'll provide more details on that aspect later.
+
+An example build definition for a Rust project called `myproject` looks like
+this:
+
+```toml
+[build.myproject]
+command = '''
+  cargo build --release
```

Comment:
this needs a `mkdir -p $out/bin` to work

Reply:
Fixed in the follow up PR from @dcarley